features/issues/components/issue-filters/issue-filters.tsx

```diff
@@ -0,0 +1,114 @@
+import { Button, Input, Option, Select } from "@features/ui";
+import styles from "./issue-filters.module.scss";
+import { useState } from "react";
+import { IssueLevel, IssueStatus } from "@api/issues.types";
+import { useRouter } from "next/router";
+
+export function IssueFilters() {
+  const [level, setLevel] = useState<Option | null>(null);
+  const [status, setStatus] = useState<Option | null>(null);
+  const [searchProject, setSearchProject] = useState<string>("");
+
+  const issueLevels = Object.values(IssueLevel).map((value) => ({
+    id: value,
+    value,
+  }));
+  const issueStatus = Object.values(IssueStatus).map((value) => ({
+    id: value,
+    value: value === "open" ? "unresolved" : value,
+  }));
+
+  const router = useRouter();
+
+  const navigateToLevel = (newLevel?: string) => {
+    if (!newLevel) {
+      const resetQuery = router.query;
+      delete resetQuery.level;
+      router.push({ pathname: router.pathname, query: resetQuery });
+      return;
+    }
+
+    router.push({
+      pathname: router.pathname,
+      query: { ...router.query, level: newLevel },
+    });
+  };
+  const navigateToStatus = (newStatus?: string) => {
+    if (!newStatus) {
+      const resetQuery = router.query;
+      delete resetQuery.status;
+      router.push({ pathname: router.pathname, query: resetQuery });
+      return;
+    }
+
+    router.push({
+      pathname: router.pathname,
+      query: { ...router.query, status: newStatus },
+    });
+  };
+  const navigateToSearchProject = (newProject?: string) => {
+    if (!newProject) {
+      const resetQuery = router.query;
+      delete resetQuery.project;
+      router.push({ pathname: router.pathname, query: resetQuery });
+      return;
+    }
+
+    router.push({
+      pathname: router.pathname,
+      query: { ...router.query, project: newProject },
+    });
+  };
```

Comment:
I might be mistaken but it looks like these could be merged into a single function!?